protocol-definitions/MC.yaml

```diff
@@ -224,6 +224,19 @@ methods:
             7 - USED_NATIVE_MEMORY_PERCENTAGE
             8 - FREE_NATIVE_MEMORY_SIZE
             9 - FREE_NATIVE_MEMORY_PERCENTAGE
+        - name: applyWanReplicationRef
+          type: boolean
+          nullable: false
+          since: 2.7
+          doc: |
+            Determines whether the supplied WanReplicationRef should be applied.
+            This is for backwards compatible versioning, as supplying a null WanReplicationRef is valid.
+        - name: wanReplicationRef
+          type: WanReplicationRef
+          nullable: true
```

Comment:
since this is nullable, do you still need the `applyWanReplicationRef` field?

Reply:
Yes, because a null `WanReplicationRef` is valid (removes the existing reference from the `MapConfig`) - so to determine if we have an actual `null` value (meaning remove the reference), or if it's just behind on protocol, the `applyWanReplicationRef` value is used (`false` in older protocols, `true` in newer). I plan to remove this value in future releases as it's just for backwards compatibility.

Reply:
So, the wanReplicationRef field is also new, and you can check existence of this new field. if the field exists(i do not mean null) then it is from new version client and you apply it. Otherwise field does not exist and it is old version client. Hence, i still could not make sense of the other field. May be i miss some context.

Reply:
I think I see what you're suggesting - so you think that [the `UpdateMapConfigOperation`](https://github.com/hazelcast/hazelcast-mono/pull/544/files#diff-60b54afbd749e10f76f9c7595e9a462d20e90f0dd73a9c8e0b99c3203fb71067) should still continue to use a `applyWanReplicationRef` field that it controls locally (if sending version is <5.4, it sets to `false`, otherwise `true`), but we don't need to send it over the wire/add to codec?

I think that could work, but please let me know if you're expecting differently.